src/app/app.component.ts

```diff
@@ -1,45 +1,100 @@
-import { Component, OnInit } from '@angular/core';
-import { Router, NavigationEnd } from '@angular/router';
+import { CookieService } from 'ngx-cookie';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Router, NavigationEnd, ActivatedRoute } from '@angular/router';
 import { UserService } from './shared/services/user.service';
 import { filter } from 'rxjs/operators';
 import { StarterService } from './shared/services/starter.service';
+import { Subscription } from 'rxjs';
 @Component({
-  selector: 'app-root',
-  templateUrl: './app.component.html',
-  styleUrls: ['./app.component.scss']
+  selector: "app-root",
+  templateUrl: "./app.component.html",
+  styleUrls: ["./app.component.scss"],
 })
-
-export class AppComponent implements OnInit {
-  title = 'civis';
+export class AppComponent implements OnInit, OnDestroy {
+  title = "civis";
   showCitySelection: boolean;
+  paramsSubscription: Subscription;
+  selectedLanguage = "en";
+  confirmMessage = {};
+  language = {
+    id: null,
+    name: null
+  };
+  isConfirmModal = false;
+  languages = [
+    {
+      id: "en",
+      name: "English",
+    },
+    {
+      id: "hi",
+      name: "Hindi",
+    },
+  ];
   constructor(
     private userService: UserService,
     private router: Router,
+    private route: ActivatedRoute,
     private starterService: StarterService,
+    private _cookieService: CookieService
   ) {
     this.starterService.loadGoogleAnalyticsSupport();
+    const currentLanguage = this._cookieService.get("civisLang");
+    if (currentLanguage) {
+      this.selectedLanguage = currentLanguage;
+    }
   }
 
   ngOnInit() {
     this.router.events.subscribe((evt) => {
       if (evt instanceof NavigationEnd) {
-         window.scrollTo(0, 0);
+        window.scrollTo(0, 0);
       }
     });
     this.checkCityPresent();
+    this.checkLang();
   }
 
+  confirmed(event: any) {
```

Comment:
@Harshwardhan18 What's this?

Reply:
@avikt it responds on how the website will behave upon selection of Ok or cancel for site translation